client/src/app/pages/dashboard/maps/map-list/map-list.component.ts

```diff
@@ -121,13 +121,19 @@ export class MapListComponent implements OnInit {
   }
 
   libraryUpdate(): void {
-    if (this.type === MapListType.TYPE_LIBRARY)
+    if (this.type === MapListType.TYPE_LIBRARY) {
+      if (this.maps.length === 1 && this.currentPage * this.pageLimit >= this.mapCount && this.currentPage > 1)
+        this.currentPage -= 1;
       this.loadMaps();
+    }
   }
 
   favoriteUpdate() {
-    if (this.type === MapListType.TYPE_FAVORITES)
+    if (this.type === MapListType.TYPE_FAVORITES) {
+      if (this.maps.length === 1 && this.currentPage * this.pageLimit >= this.mapCount && this.currentPage > 1)
```

Comment:
I'd like to see this duplicated check: `this.maps.length === 1 && this.currentPage * this.pageLimit >= this.mapCount && this.currentPage > 1` be moved to a function with a descriptive name